### ISSUE TITLE
problem with label reading

### ISSUE THREAD

**Issue by gaopeng-eugene** [DrSleep/tensorflow-deeplab-resnet#43]
I follow your code and write a program to read the labels for single image. However the label is weired.

atplotlib inline
import matplotlib.pyplot as plt
import tensorflow as tf
import numpy as np

label_name = tf.placeholder(tf.string,None)
label = tf.image.decode_png(tf.read_file(label_name), channels=1)
sess = tf.Session()

output = sess.run(label,feed_dict={label_name:'/opt/data/penggao/dataset/VOCdevkit/VOC2012/SegmentationClass/2010_001131.png'})
np.unique(output)
array([  0,  52, 132, 147, 220], dtype=uint8)

**Comment by gaopeng-eugene** [comment]
what is the problem with the code? When I use your code, everything is right.

**Comment by gaopeng-eugene** [comment]
Can you tell me where is go wrong? I am doing experiments on COCO in which I need to know how to arrange the label.

**Comment by jia2lin3yuan1** [comment]
I met the same issue too. One way for it is to convert label images into real gray level png files before call the model, like using scipy.misc.imread(fileName, mode = 'P'). Hope some guy could provide solutions inside tensorflow.

**Comment by gaopeng-eugene** [comment]
But when you output the label in train.py or evaluate.py. The results is right. It's very weired.

**Comment by jia2lin3yuan1** [comment]
Output is right, because the pretrained model is correct. Since the label input is not correct, there is no training on your specific dataset.

**Comment by DrSleep** [comment]
@gaopeng-eugene, nothing is wrong with the code. If you use SegmentationClassAug instead of SegmentationClass, the output will be correct.

As for the difference between Class and ClassAug, the latter one does not contain the palette (the colour map). From the [DeepLab FAQ page](http://ccvl.stat.ucla.edu/deeplab_faq/):

> Q: The label read by IMAGE_SEG_DATA is not the same as expected. Do you pre-process the annotations?
> 
> A: You need to save the png annotations without the colormap; otherwise OpenCV will read the wrong values. See the script at matlab/my_script/SavePngAsRawPng.m for reference.
